core/eds.go

```diff
@@ -28,7 +28,7 @@ import (
 // nil is returned in place of the eds.
 func extendBlock(data types.Data, appVersion uint64, options ...nmt.Option) (*rsmt2d.ExtendedDataSquare, error) {
 	if app.IsEmptyBlock(data, appVersion) {
-		return nil, nil
+		return share.EmptyExtendedDataSquare(), nil
```

Comment:
This might introduce the bug, pls check how the code is being used

Reply:
I agree. There seem to be no special handling for nil eds further in the code. Need @renaynay to confirm

Reply:
All the tests are broken, so likely related

Reply:
Just flack, unrelated. 
```
discovery_test.go:134: did not discover peer in time
```
https://github.com/celestiaorg/celestia-node/actions/runs/8783227354/job/24099029082?pr=3332#step:4:87

Reply:
https://github.com/celestiaorg/celestia-node/pull/3332/commits/d97a9d359bc178677c758e0b34354b418af02f23